processing/src/main/java/org/apache/druid/segment/column/ComplexColumn.java

```diff
@@ -19,45 +19,34 @@
 
 package org.apache.druid.segment.column;
 
+import org.apache.druid.guice.annotations.ExtensionPoint;
 import org.apache.druid.query.monomorphicprocessing.RuntimeShapeInspector;
 import org.apache.druid.segment.ColumnValueSelector;
 import org.apache.druid.segment.ObjectColumnSelector;
-import org.apache.druid.segment.data.GenericIndexed;
 import org.apache.druid.segment.data.ReadableOffset;
 
 import javax.annotation.Nullable;
 
 /**
-*/
-public class ComplexColumn implements BaseColumn
+ * This interace represents a complex column and can be implemented by druid extension writer of a custom column
+ * with arbitrary serialization instead of a custom column that serializes rows of objects serialized using
+ * {@link org.apache.druid.segment.data.GenericIndexed} class which is default implementation of "writeToXXX" methods in
+ * {@link org.apache.druid.segment.serde.ComplexColumnSerializer}. In that case {@link GenericIndexedBasedComplexColumn}
+ * should be used.
+ */
+@ExtensionPoint
```

Comment:
Do additional column interfaces need to be marked as @ExtensionPoint for this to make sense?

Reply:
that is good point. I did another pass at things and most of the stuff looks to be marked correctly already. I had to mark ColumnValueSelector hierarchy as @ExtensionPoint .